homeassistant/components/nightscout/sensor.py

```diff
@@ -28,10 +30,24 @@ async def async_setup_entry(
     entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
-    """Set up the Glucose Sensor."""
-    api = hass.data[DOMAIN][entry.entry_id]
+    """Set up entities."""
+    data = hass.data[DOMAIN][entry.entry_id]
+    api = data[API]
+    coordinator = data[COORDINATOR]
+
+    # Glucose sensor
     async_add_entities([NightscoutSensor(api, "Blood Sugar", entry.unique_id)], True)
 
+    # Uploader batteries
+    entities = []
+
+    for device_name in coordinator.data:
+        entities.append(
+            Battery(coordinator, device_name, f"{entry.unique_id}_{device_name}")
+        )
+
+    async_add_entities(entities)
```

Comment:
Please collect all entities first in a list and then make a single call to `async_add_entities`.

Reply:
Fixed!